cudax/include/cuda/experimental/__execution/queries.cuh

```diff
@@ -63,18 +63,20 @@ using _CUDA_STD_EXEC::__queryable_with;
 _CCCL_GLOBAL_CONSTANT struct get_allocator_t
 {
   _CCCL_EXEC_CHECK_DISABLE
-  template <class _Env>
-  [[nodiscard]] _CCCL_API auto operator()(const _Env& __env) const noexcept
+  _CCCL_TEMPLATE(class _Env)
+  _CCCL_REQUIRES(__queryable_with<_Env, get_allocator_t>)
+  [[nodiscard]] _CCCL_API constexpr auto operator()(const _Env& __env) const noexcept
+    -> __query_result_t<_Env, get_allocator_t>
```

Comment:
I am really unhappy with this. 

Would it make sense to introduce a `__query_result_or_t` alias that boils down to

```cpp
template<class _Env, class _Query, class _Fallback>
using __query_result_or_t = _CUDA_VSTD::_If<__queryable_with<_Env, _Query>, __query_result_t<_Env, _Query>, _Fallback>;
```

Reply:
this is a great suggestion! there already is a `__query_result_or_t` alias, and a `__query_or` helper function.